scipy/_lib/_array_api.py

```diff
@@ -490,6 +490,15 @@ def xp_ravel(x: Array, /, *, xp: ModuleType | None = None) -> Array:
     return xp.reshape(x, (-1,))
 
 
+def xp_swapaxes(a, axis1, axis2, xp=None):
+    # Equivalent of np.swapaxes written in terms of array API
+    xp = array_namespace(a) if xp is None else xp
+    axes = list(range(a.ndim))
+    axes[axis1], axes[axis2] = axes[axis2], axes[axis1]
+    a = xp.permute_dims(a, axes)
+    return a
```

Comment:
potential contribution to array-api-extra? Might be good to at least open an issue.

Reply:
cross-ref https://github.com/scipy/scipy/blob/main/scipy/signal/_signaltools.py#L845

Reply:
I forgot that there is https://github.com/data-apis/array-api-extra/issues/97

Reply:
Could we use `moveaxis(a, (axis2, axis1), (axis1, axis2))` to simplify this implementation (or just get rid of the helper)?

Reply:
Oops, I wrote this before `moveaxis` existed.

Reply:
I tried, and it's not as simple as that:
```
x = xp.moveaxis(x, (axis, -1), (-1, axis))
```
fails on all backends when `axis` is the same as axis `-1`. 

If you prefer, feel free to commit:
```suggestion
def xp_swapaxes(a, axis1, axis2, xp=None):
    # Equivalent of np.swapaxes written in terms of array API
    xp = array_namespace(a) if xp is None else xp
    axis1 = a.ndim + axis1 if axis1 < 0 else axis1
    axis2 = a.ndim + axis2 if axis2 < 0 else axis2
    if axis1 == axis2:
        return a
    return xp.moveaxis(a, (axis1, axis2), (axis2, axis1))
```

Reply:
actually, maybe this is an improvement?

```suggestion
def xp_swapaxes(a, axis1, axis2, xp=None):
    # Equivalent of np.swapaxes written in terms of array API
    xp = array_namespace(a) if xp is None else xp
    ndim = a.ndim
    if (axis1 % ndim) == (axis2 % ndim):
        return a
    return xp.moveaxis(a, (axis1, axis2), (axis2, axis1))
```

Reply:
Feel free to implement in xpx as you see fit.

Reply:
Actually the modulos with walrus are pretty clever; I didn't think of that. It brings to mind that both of these will not raise with invalid axes in some cases, though.